### DISCUSSION TITLE
Bug reports and improvement suggestions

### DISCUSSION THREAD

**Discussion by wj-qiao** [sumatrapdfreader/sumatrapdf#3736] [General]
Hi there,

I'm using the latest pre-release version of SumatraPDF, the version is v3.5.15697 64 bit on Win10,

bug report:
The free text written in Chinese is missing characters, for example, the comments of my annotation is "为什么是这样的？", while the text shown in the PDF is "什么是的？", some characters are missing, while the free text written in English is fine in show.

improvement suggestion:
The "Contents" of the free text is default as "This is a text...",  is it possible to leave the "Contents" of free text blank, or alternatively, set it as an optional field with the ability to customize the default content?

BTW, I really enjoy using SumatraPDF, and I hope it becomes even more user-friendly.

**Comment by wj-qiao** [comment]
Additionally,

improvements suggestion:
The free text boxes cannot be resized, which may lead to annotations covering the original content.

**Comment by carculator** [comment]
Issue with ALL Versions: Sumatra PDF is showing a "Code 39" Barcode in plain letters whereas Adobe Reader is showing it correctly.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/45626080/610767b9-c06c-4090-8c0a-1fd972ec8fb9)
Here is Acrobat:
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/45626080/6bb2db9d-71ec-40fa-99fa-33b2440ce7c3)
Fontinfos from Sumatra:
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/45626080/b2fa4c07-f9fb-4722-9f99-cd619975e250)
[Seiten aus MAPA_C65_679_#0101_AUFTRAG-DZEWienVStVAUSLRS_ohne_Signatur-output-20230927T010103_2ed7d59632d9444099f12b26c11e2301.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/12735972/Seiten.aus.MAPA_C65_679_.0101_AUFTRAG-DZEWienVStVAUSLRS_ohne_Signatur-output-20230927T010103_2ed7d59632d9444099f12b26c11e2301.pdf)
[CODE39.zip](https://github.com/sumatrapdfreader/sumatrapdf/files/12735977/CODE39.zip)

Although Sumatra does not display the text correctly, it prints it as a barcode.

**Reply by GitHubRulesOK** [reply]
When a font is not imbedded in the file then Acrobat in Edge shows it like this as AdobeSansMM
that font is not the one listed but Adobe will use it to show the characters,
since it cannot decipher those imbedded binary characters without the font encoding.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/52d4f6c4-a174-4386-8f2e-17004f5ff4e8)

**Reply by carculator** [reply]
Acrobat DOES display the correct visual conversion as a barcode, only sumatra does not.
Why embed the fonts if they are reliably installed on the used computer?

**Reply by GitHubRulesOK** [reply]
SumatraPDF can only show such codes when embeded not described as ANSI which **Acrobat enabled Edge** also shows as it does not have a library to draw from.
Here I used the font in Word which embeds the graphics not just the ANSI codes
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/d4770ffe-460c-4396-84ae-bbf5f2322803)
Here is what Acrobat sees
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/44a9bc01-a8c2-421f-b8c1-53bd141465ee)


There are two levels of encoding in PDF the raw data is just 1 and 0 s that is translated into ANSI by the reader seen on the left

if a font is imbedded then those ANSI codes are applied to a look up table and converted as a different vector graphic from the font. 
If there is no embedment the system may be searched for a substitute font usually System Default of Arial or if user has exact font that may be used by font substitution engine (this what full Acrobat Reader does but not lighter Acrobat Powered Edge.)
SumatraPDF and most other readers do not carry a large set of fonts to draw from.

The Font is not seen by other non Adobe systems it has to be forced to something suitable
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/17509dd4-e7ce-447d-9f7b-321b7eb56812)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/39ef5751-f41a-4e02-a1da-07cef258c159)


Here is Raw binary PDF converted to Human ANSI as seen by **Acrobat and all other readers**
So point to note readers are simply given a string of letters and a reference a font of `/F0` is to be applied at 24 points high
```
BT
24 TL
/F0 24 Tf
1 0 0 1 326 810 Tm
(*$A9$BDB1A*) Tj
36 TL
/F1 36 Tf
1 0 0 1 560 810 Tm
(T) Tj
11 TL
/F1 11 Tf
1 0 0 1 16 810 Tm
(776986) Tj
```
lets see what F0 hints at = `/Font<</F0 10 0 R`
```
10 0 obj
<</Subtype/TrueType/FontDescriptor 14 0 R...Code#2039/FirstChar 0/Encoding/WinAnsiEncoding/Type/Font
```
leads to
```
14 0 obj
<</StemV 132/FontName/Code#2039
```
so what font name can that be ? I tell you what, lets use the one we know is ansi Times Roman or Helvetica should be good enough for humans to read, but the hinting suggest a Sans Serif font should do.

**Reply by carculator** [reply]
Found the answer: one must install the font "_for all users_" 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/45626080/4aff9f09-f0fa-43bf-9029-177b1b1fc0c4)
Is Sumatra PDF running within another user space context ?

**Reply by carculator** [reply]
But thank you anyway for the clearings
